libs/nss/Makefile

```diff
@@ -0,0 +1,104 @@
+#
+# Copyright (C) 2017 Lucian Cristian
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=nss
+PKG_VERSION:=3.33
+PKG_RELEASE:=1
+PKG_MAINTAINER:=Lucian Cristian <lucian.cristian@gmail.com>
+PKG_LICENCE:=GPL-2.0+
+
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION)-with-nspr-4.17.tar.gz
+PKG_SOURCE_URL:=https://ftp.mozilla.org/pub/security/nss/releases/NSS_3_33_RTM/src/
+PKG_HASH:=878d505ec0be577c45990c57eb5d2e5c8696bfa3412bd0fae193b275297bf5c4
+
+PKG_BUILD_PARALLEL:=1
+
+PKG_FIXUP:=libtool
+
+include $(INCLUDE_DIR)/package.mk
+
+x64=--enable-64bit
+
+ifeq ($(ARCH),aarch64)
+	conf=$(x64)
+else ifeq ($(ARCH),ppc64)
+	conf=$(x64)
+else ifeq ($(ARCH),x86_64)
+	conf=$(x64)
+endif
+
+define Build/Compile
+	(cd $(PKG_BUILD_DIR)/nss; \
+		NSPR_CONFIGURE_OPTS="--build=$(GNU_HOST_NAME) \
+				     --host=$(GNU_HOST_NAME) \
+				     --target=$(REAL_GNU_TARGET_NAME) \
+				     --disable-debug \
+				     --enable-optimize \
+				     --enable-strip \
+				     $(if $(CONFIG_IPV6),--enable-ipv6,) \
+				     $(conf)" \
+		make nss_build_all NSS_ENABLE_WERROR=0 BUILD_OPT=1 CROSS_COMPILE=1 \
+		OS_TARGET=OpenWRT OBJDIR_NAME=build_dir NSS_USE_SYSTEM_SQLITE=1 NSS_DISABLE_GTESTS=1 SEED_ONLY_DEV_URANDOM=1 \
+		TARGET=$(TARGET_CC) TARGETCXX=$(TARGET_CXX) CPU_ARCH=$(ARCH) MUSL=$(if $(CONFIG_LIBC_USE_GLIBC),0,1))
+endef
+
+define Package/libnss
+  SECTION:=libs
+  SUBMENU:=SSL
+  CATEGORY:=Libraries
+  TITLE:=Library Network Security Services (NSS)
+  URL:=http://www.mozilla.org/projects/security/pki/nss/
+  DEPENDS:=+libpthread +librt +zlib +libstdcpp +libsqlite3
```

Comment:
is uclibc++ possible here?

Reply:
I think is not needed anymore but I got a linking problem with libreswan that I didn't saw last time, need more testing

Reply:
Ah the catch is that everything must include uclibc++.mk. you can't link between uclibc++ and libstdcpp.

Reply:
For example this: https://github.com/neheb/packages/commit/33b1429019b0182c84d3ec3d8a1e10a4c65bc09e

Only ibrcommon depends on any kind of libc++ (technically they all do) but uclibc++.mk must still be included in all of them.